sdk/core/core-client-rest/src/sendRequest.ts

```diff
@@ -27,55 +31,130 @@ export async function sendRequest(
+    ...(options.headers ? options.headers : {}),
     accept: options.accept ?? "application/json",
-    ...(body !== undefined && {
+    ...(hasContent && {
       "content-type": options.contentType ?? getContentType(options.body),
     }),
-    ...(options.headers ? options.headers : {}),
   });
 
-  const request = createPipelineRequest({
+  return createPipelineRequest({
     url,
     method,
     body,
+    formData,
     headers,
     allowInsecureConnection: options.allowInsecureConnection,
   });
+}
+
+interface RequestBody {
+  body?: RequestBodyType;
+  formData?: FormDataMap;
+}
 
-  const result = await pipeline.sendRequest(httpClient, request);
-  const rawHeaders: RawHttpHeaders = result.headers.toJSON();
+/**
+ * Prepares the body before sending the request
+ */
+function getRequestBody(body?: unknown, contentType: string = "application/json"): RequestBody {
+  if (body === undefined) {
+    return { body: undefined };
+  }
 
-  let parsedBody = undefined;
+  const firstType = contentType.split(";")[0];
 
-  try {
-    parsedBody = result.bodyAsText ? JSON.parse(result.bodyAsText) : undefined;
-  } catch {
-    parsedBody = undefined;
+  switch (firstType) {
+    case "multipart/form-data":
+      return isFormData(body) ? { formData: body } : { body: JSON.stringify(body) };
+    case "text/plain":
+      return { body: String(body) };
+    default:
+      return { body: JSON.stringify(body) };
   }
+}
 
-  return {
-    request,
-    headers: rawHeaders,
-    status: `${result.status}`,
-    body: parsedBody,
-  };
+function isFormData(body: unknown): body is FormDataMap {
+  return body instanceof Object && Object.keys(body).length > 0;
 }
 
 /**
- * Function to determine the content-type of a body
- * this is used if an explicit content-type is not provided
- * @param body - body in the request
- * @returns returns the content-type
+ * Prepares the response body
  */
-function getContentType(body: any): string {
-  if (ArrayBuffer.isView(body)) {
-    return "application/octet-stream";
+function getResponseBody(response: PipelineResponse): RequestBodyType | undefined {
+  // Set the default response type
+  const contentType = response.headers.get("content-type") ?? "";
+  const firstType = contentType.split(";")[0];
+  const bodyToParse: string = response.bodyAsText ?? "";
+
+  if (firstType === "text/plain") {
+    return String(bodyToParse);
   }
 
-  // By default return json
-  return "application/json; charset=UTF-8";
+  // Default to "application/json" and fallback to string;
+  try {
+    return bodyToParse ? JSON.parse(bodyToParse) : undefined;
+  } catch (error) {
+    // If we were supposed to get a JSON object and failed to
+    // parse, throw a parse error
+    if (firstType === "application/json") {
+      throw createParseError(response, error);
+    }
+
+    // We are not sure how to handle the response so we return it as
+    // plain text.
+    return String(bodyToParse);
```

Comment:
Did you consider returning it as is? it could be valuable to return it as is if it were bytes array/binary.

Reply:
Good question! I'm about to send out a PR to explicitly handle binary 😄